package/MDAnalysis/core/universe.py

```diff
@@ -1459,13 +1482,101 @@ def from_smiles(cls, smiles, sanitize=True, addHs=True,
 
         return cls(mol, **kwargs)
 
+    def guess_TopologyAttrs(
+            self, context=None, to_guess=(), force_guess=(), **kwargs):
+        """
+        Guess and add attributes through a specific context-aware guesser.
+
+        Parameters
+        ----------
+        context: str or Guesser class
+            For calling a matching guesser class for this specific context
+        to_guess: list[str], (optional), default ``['types', 'masses']``
+                (in future versions types and masses will be removed)
+            TopologyAttrs to be guessed. These TopologyAttrs will be wholly guessed
+            if they don't exist in the universe. If they already exist in the Universe,
+            only empty or missing values will be guessed.
+        force_guess: list[str], (optional)
+            TopologyAttrs in this list will be force guessed. If the TopologyAttr
+            does not already exist in the Universe, this has no effect. If the TopologyAttr
+            does already exist, all values will be overwritten by guessed values.
+        **kwargs: extra arguments to be passed to the guesser class
+
+        Examples
+        --------
+        guess ``masses`` and ``types`` attribute::
+
+            u.guess_TopologyAttrs(context='default', to_guess=['masses', 'types'])
+
+        .. versionadded:: 2.8.0
+
+        """
+        if not context:
+            context = self._context
+
+        guesser = get_guesser(context, self.universe, **kwargs)
+        self._context = guesser
+
+        total_guess = list(force_guess) + list(to_guess)
+
+        # Removing duplicates from the guess list while keeping attributes
+        # order as it is more convientent to guess attributes
+        # in the same order that the user provided
+        total_guess = list(dict.fromkeys(total_guess))
+
+        objects = ['bonds', 'angles', 'dihedrals', 'impropers']
+
+        # Checking if the universe is empty to avoid errors
+        # from guesser methods
+        if self._topology.n_atoms > 0:
+
+            topology_attrs = [att.attrname for att in
+                              self._topology.read_attributes]
+
+            common_attrs = set(to_guess) & set(topology_attrs)
+            common_attrs = ", ".join(attr for attr in common_attrs)
+
+            if len(common_attrs) > 0:
+                logger.info(
+                    f'The attribute(s) {common_attrs} have already been read '
+                    'from the topology file. The guesser will '
+                    'only guess empty values for this attribute, '
+                    'if any exists. To overwrite it by completely '
+                    'guessed values, you can pass the attribute to'
+                    ' the force_guess parameter instead of '
+                    'the to_guess one')
+
+            for attr in total_guess:
+                if guesser.is_guessable(attr):
+                    fg = True if attr in force_guess else False
```

Comment:
Can you write this just as
```suggestion
                    fg = attr in force_guess
```